playwright/fixtures/makeName/generators/generateLegacyNameWithConfig.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable import/no-extraneous-dependencies */
+
+/* eslint-disable no-await-in-loop */
+import { Accounts, User } from 'playwright/fixtures/accounts'
+import { Contracts } from 'playwright/fixtures/contracts'
+
+import { namehash } from '@ensdomains/ensjs/utils/normalise'
+import { RecordOptions } from '@ensdomains/ensjs/utils/recordHelpers'
+
+import { RESOLVER_ADDRESSES } from '@app/utils/constants'
+
+import { Provider } from '../../provider'
+import { LegacySubname, generateLegacySubname } from './generateLegacySubname'
+import { generateRecords } from './generateRecords'
+
+const DEFAULT_DURATION = 31536000
+const DEFAULT_RESOLVER = RESOLVER_ADDRESSES['1337'][2] as `0x${string}`
+const VALID_RESOLVERS = RESOLVER_ADDRESSES['1337'].filter(
+  (resolver) => resolver !== '0xd7a4F6473f32aC2Af804B3686AE8F1932bC35750',
+)
+
+export type Name = {
+  label: string
+  owner?: User
+  manager?: User
+  duration?: number
+  secret?: string
+  resolver?: `0x${string}`
+  addr?: User
+  records?: RecordOptions
+  subnames?: Omit<LegacySubname, 'name'>[]
+}
+
+type Dependencies = {
+  provider: Provider
+  accounts: Accounts
+  contracts: Contracts
+}
+
+export const generateLegacyNameWithConfig =
+  ({ provider, accounts, contracts }: Dependencies) =>
+  async ({
+    label,
+    owner = 'user',
+    manager,
+    duration = DEFAULT_DURATION,
+    // eslint-disable-next-line no-restricted-syntax
+    secret = '0x0000000000000000000000000000000000000000000000000000000000000000',
+    resolver = DEFAULT_RESOLVER,
+    addr = owner,
+    records,
+    subnames,
+  }: Name) => {
+    const name = `${label}.eth`
+    console.log('generating legacy name:', name)
+
+    const _owner = accounts.getAddress(owner)
+    const _addr = accounts.getAddress(addr)
+
+    // Registration will fail if resoler is not valid. If an invalid resolver is entered
+    // we will set the resolver after the name has been registered.
+    const hasValidResolver = resolver && VALID_RESOLVERS.includes(resolver)
+    const _resolver = hasValidResolver ? resolver : DEFAULT_RESOLVER
+
+    console.log('making commitment:', name)
+    const controller = contracts.get('LegacyETHRegistrarController', { signer: owner })
+    const commitment = await controller.makeCommitmentWithConfig(
+      label,
+      _owner,
+      secret,
+      _resolver,
+      _addr,
+    )
+    const commitTx = await controller.commit(commitment)
+    await commitTx.wait()
+
+    await provider.increaseTime(120)
```

Comment:
When calling increaseTime, sometimes it wouldn't increase by all 60s, being off by a couple seconds. I checked by printing out time before and after increaseTime, so I doubled the time for safety.

Reply:
is there a deeper problem there then? if you've already looked into it then this works fine but otherwise there could be other issues that stem from something similar